src/Maui/Features/Creatures/Import/Upload.razor.cs

```diff
@@ -11,11 +13,14 @@ private async Task UploadFilesAsync(IReadOnlyList<IBrowserFile> files)
     {
         foreach (var file in files)
         {
-            var creature = await JsonSerializer.DeserializeAsync<Creature>(file.OpenReadStream(512000, default), new JsonSerializerOptions
+            var tetraCubeCreature = await JsonSerializer.DeserializeAsync<TetraCubeCreature>(file.OpenReadStream(512000, default), new JsonSerializerOptions
             {
                 PropertyNameCaseInsensitive = true
             });
-
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<TetraCubeCreature, Creature>());
```

Comment:
We’re configuring this through reflection. Look in Core for some profile examples.